### DISCUSSION TITLE
Create action when 'clicking' on a record displayed in a table

### DISCUSSION THREAD

**Discussion by GonnaDropU** [Budibase/budibase#4532] [Help]
Hey Friends, SUPER Noob here so please be gentle.  I am working to help a school district create a way for very handicapped children to work on the ability to enter their 6 digit lunch ID into the system.  These kiddos are severely handicapped.  I have created a screen for viewing student data, a screen for entering 'deleting students and student IDs into the db (hosted by budbase) I have also created a "select your name from the list" page that does display the students' names.  Using Data Provider and table
From here I want the teacher, or student, to be able to click select the name and go to a new screen.
On the new screen, it might say "WELCOME student_name, PLEASE ENTER YOUR STUDENT ID" with a text box.  They will use the 10 key to type in their 6 digit id and enter/submit.  I want to validate what they entered against the student id field in the db and (1) congrats screen or (2) try again screen.  
Is something like this even possible with this or am I really pushing it... ie. I'm no coder

**Comment by mjashanks** [comment]
Hey @GonnaDropU - I was able to successfully build this, but I needed a bit of Javascript. The app is attached below.

How it's done
1. I have the structure
```
> Form 
   > Text Field (studentId search)
   > Data Provider : Filter on StudentID = Text Field
       > Repeater
            > Headline: Thank you!
       > Headline: Not found - shows conditionally
```

2. The Data Provider filters on the value from the text field. Javascript is used in here to filter on either the value on the text field, or "9999999999" if the field is empty - i.e. a value that will return no results!
3. The "Not Found" headline only displays if there is a value in the studentId search and zero rows have been returned by the data provider

My App: [App-export-1645460120069.txt](https://github.com/Budibase/budibase/files/8110755/Michaels.App-export-1645460120069.txt)
